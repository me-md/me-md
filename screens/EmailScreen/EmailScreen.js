import React, { Fragment, useState } from 'react';
import { Dimensions, Image, StyleSheet, View } from 'react-native';
import { Button, Input, Item, Text } from 'native-base';
import { Header } from '../../components/Header';
import { Ionicons, MaterialCommunityIcons } from '@expo/vector-icons';
import { AntDesign } from '@expo/vector-icons';
import * as EmailValidator from 'email-validator';
import { sendEmailReport } from '../../utils/apiCalls/SendEmail/sendEmailReport';

const height = Dimensions.get('window').height;

export default function EmailScreen({ navigation }) {
  const { location, report, stateAbbreviation } = navigation.state.params;
  const [email, setEmail] = useState('');
  const [emailSent, setEmailSent] = useState(false);
  const [enabled, setEnabled] = useState(false);

  const sendEmail = async () => {
    setEmailSent(true);
    return await sendEmailReport(email, report);
  };

  const validateEmail = email => {
    EmailValidator.validate(email) && setEnabled(true);
    setEmail(email);
  };

  return (
    <Fragment>
      <Header />
      <AntDesign
        name='close'
        size={24}
        color='black'
        style={styles.close}
        onPress={() =>
          navigation.push('Doctors', { location, stateAbbreviation })
        }
      />
      <View style={styles.container}>
        <Text style={styles.title}>Send Detailed Report:</Text>
        <Image
          source={require('../../assets/images/report.png')}
          style={styles.report}
        />
        {emailSent ? (
          <Fragment>
            <Text style={styles.response}>Thank you!</Text>
            <Text
              style={styles.thankYou}
            >{`A detailed report from today's session has been sent to: ${email}`}</Text>
            <Button
              rounded
              style={styles.button}
              onPress={() => navigation.push('Welcome')}
            >
              <Text style={styles.buttonText}>New Checkup</Text>
              <MaterialCommunityIcons
                name='stethoscope'
                style={styles.icon}
                size={26}
                color='white'
              />
            </Button>
          </Fragment>
        ) : (
            <Fragment>
              <Item style={styles.input}>
                <Input
                  placeholder='Enter your email address'
                  style={styles.input}
                  autoCapitalize='none'
                  onChangeText={text => validateEmail(text)}
                />
              </Item>
              <View style={styles.buttonContainer}>
                {enabled ? (
                  <Button
                    rounded
                    style={styles.button}
                    onPress={() => sendEmail()}
                  >
                    <Text style={styles.buttonText}>Send Report</Text>
                    <Ionicons
                      name='ios-send'
                      style={styles.icon}
                      size={26}
                      color='white'
                    />
                  </Button>
                ) : (
                    <Button
                      disabled
                      rounded
                      style={styles.button}
                      onPress={() => sendEmail()}
                    >
                      <Text style={styles.buttonText}>Send Report</Text>
                      <Ionicons
                        name='ios-send'
                        style={styles.icon}
                        size={26}
                        color='white'
                      />
                    </Button>
                  )}
                <Button
                  rounded
                  style={styles.button}
                  onPress={() => navigation.push('Welcome')}
                >
                  <Text style={styles.buttonText}>New Checkup</Text>
                  <MaterialCommunityIcons
                    name='stethoscope'
                    style={styles.icon}
                    size={26}
                    color='white'
                  />
                </Button>
              </View>
            </Fragment>
          )}
      </View>
    </Fragment>
  );
}

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    flex: 1,
    justifyContent: 'space-between',
    paddingBottom: height * 0.075,
    width: '100%'
  },
  contentContainer: {
    alignItems: 'center',
    flex: 1
  },
  close: {
    alignSelf: 'flex-end',
    marginRight: height * 0.02,
    marginTop: height * 0.025
  },
  input: {
    fontSize: 16,
    textAlign: 'center',
    width: '60%'
  },
  title: {
    fontSize: 36,
    fontWeight: 'bold',
    paddingTop: height * 0.025
  },
  button: {
    alignItems: 'center',
    justifyContent: 'space-around',
    marginTop: height * 0.02,
    width: '60%'
  },
  buttonText: {
    fontSize: 20,
    fontWeight: 'bold'
  },
  buttonContainer: {
    marginBottom: height * 0.08
  },
  icon: {
    marginRight: height * 0.02
  },
  report: {
    alignSelf: 'center',
    borderWidth: 0,
    height: height * 0.17,
    marginBottom: height * 0.01,
    marginTop: height * 0.04,
    width: height * 0.17
  },
  response: {
    fontSize: 24,
    fontWeight: 'bold'
  },
  thankYou: {
    marginLeft: height * 0.02,
    marginRight: height * 0.02
  }
});
