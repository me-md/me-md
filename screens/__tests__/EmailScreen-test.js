import React from 'react';
import NavigationTestUtils from 'react-navigation/NavigationTestUtils';
import renderer from 'react-test-renderer';

import EmailScreen from '../../screens/EmailScreen/EmailScreen';

jest.mock('expo', () => ({
  AppLoading: 'AppLoading',
}));



describe('EmailScreen', () => {
  jest.useFakeTimers();

  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it.skip(`should match a snapshot`, () => {
    const mockNavigation = {

      state: {
        params: {
          location: {
            lat: 121,
            long: 25
          },
          report: {
            id: 1,
            age: 22
          },
          stateAbbreviation: 'CO'
        }
      }
    };

    const tree = renderer.create(<EmailScreen navigation={mockNavigation} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

});
