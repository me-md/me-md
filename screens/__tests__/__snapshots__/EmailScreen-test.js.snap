// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailScreen should match a snapshot 1`] = `
Array [
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          4278210303,
          4287155441,
        ],
        "endPoint": undefined,
        "locations": undefined,
        "startPoint": undefined,
      }
    }
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 200.1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "alignSelf": "center",
          "borderWidth": 0,
          "height": 133.4,
          "marginBottom": 13.34,
          "marginTop": 60.03,
          "width": 133.4,
        }
      }
    />
  </ViewManagerAdapter_ExpoLinearGradient>,
  <Text />,
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-between",
        "paddingBottom": 100.05,
        "width": 750,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
            "fontFamily": "System",
            "fontSize": 16,
          },
          Object {
            "fontSize": 52.50000000000001,
            "fontWeight": "bold",
            "paddingTop": 13.34,
            "textAlign": "center",
            "width": 675,
          },
        ]
      }
      uppercase={false}
    >
      Send Detailed Report
    </Text>
    <Image
      source={1}
      style={
        Object {
          "alignSelf": "center",
          "borderWidth": 0,
          "height": 226.78000000000003,
          "marginBottom": 13.34,
          "marginTop": 40.019999999999996,
          "width": 226.78000000000003,
        }
      }
    />
    <View
      accessible={true}
      focusable={false}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": 1,
          "borderColor": "#D9D5DC",
          "borderLeftWidth": 0,
          "borderRadius": undefined,
          "borderRightWidth": 0,
          "borderTopWidth": 0,
          "borderWidth": undefined,
          "flexDirection": "row",
          "marginBottom": 133.4,
          "marginLeft": 2,
          "opacity": 1,
          "width": 600,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        editable={true}
        onChangeText={[Function]}
        placeholder="Enter your email address"
        placeholderTextColor="#575757"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "color": "#000",
              "flex": 1,
              "fontSize": 17,
              "height": 50,
              "paddingLeft": 5,
              "paddingRight": 5,
              "top": 1.5,
            },
            Object {
              "fontSize": 30,
              "textAlign": "center",
              "width": 450,
            },
          ]
        }
        underlineColorAndroid="rgba(0,0,0,0)"
      />
    </View>
    <View
      style={
        Object {
          "marginBottom": 106.72,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#b5b5b5",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 57,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 2,
            "flexDirection": "row",
            "height": 93.38000000000001,
            "justifyContent": "space-around",
            "marginTop": 66.7,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 5,
              "width": 5,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 4.65,
            "width": 487.5,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#fff",
                "fontFamily": "System",
                "fontSize": 16.5,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              },
              Object {
                "fontSize": 33.35,
                "fontWeight": "bold",
              },
            ]
          }
          uppercase={false}
        >
          Send Report
        </Text>
        <Text />
      </View>
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#007aff",
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 57,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 2,
            "flexDirection": "row",
            "height": 93.38000000000001,
            "justifyContent": "space-around",
            "marginTop": 66.7,
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 5,
              "width": 5,
            },
            "shadowOpacity": 0.3,
            "shadowRadius": 4.65,
            "width": 487.5,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "color": "#fff",
                "fontFamily": "System",
                "fontSize": 16.5,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              },
              Object {
                "fontSize": 33.35,
                "fontWeight": "bold",
              },
            ]
          }
          uppercase={false}
        >
          New Checkup
        </Text>
        <Text />
      </View>
    </View>
  </View>,
]
`;
